### ISSUE TITLE
html2canvas scaling - images low quality

### ISSUE THREAD

**Issue by stoffemi** [niklasvh/html2canvas#1376]
I'm using the currently latest html2canvas v1.0.0-alpha.9 with the scale option.

Im really glad this option was added.

But the problem is that the quality of images inside the output image are low quality. The source images used is high resolution and should be able to render smoothly with the given scale.

I put together a jsfiddle to illustrate the problem.

See https://jsfiddle.net/10wyrk9q/1/

Is there anyway around this problem?

**Comment by balkarov** [comment]
I have the same problem. How to disable parent scale when caprute node?

**Comment by nettiopsu** [comment]
Yes, absolutely true. The "funny" fact here is that actually text quality improves if you are using scale factor of 2. But images get worse.

In case of PDF generating I see atm problem workaround as generating texts and images separately with different scale factors (1 and 2)

**Comment by balkarov** [comment]
@nettiopsu look my [issue](https://github.com/niklasvh/html2canvas/issues/1448). It helped me

**Comment by jgritten** [comment]
any new information on this? I'm bumping into this

**Comment by mojimi** [comment]
Bumping, same problem

**Comment by idjitjohn** [comment]
Not a real solution from html2canvas, but I did this:
- scaling the original div with css:  #`transform: scale(2)`
- create the canvas
- then put back the scale to 1.
With jQuery it will be like:
`
div.css("transform", "scale(2)");
html2canvas(div).then(canvas => {
  $("body").append(canvas)
  div.css("transform", "");
}

`

**Comment by gameunite** [comment]
Any workaround for this? I tried what @idjitjohn suggested but it didn't work for me.

**Comment by FoxiMaxi** [comment]
Here is my workaround:

[](https://jsfiddle.net/05mzrc41/1/)

I used the HTML <img> tag as a background image instead of the CSS background-image:

`<div id="content">
  <img src="..." />
  <p>some text...</p>
</div>

<style>
       ...
      #content {
        display: inline-block;
        overflow: hidden;
        position: relative;
        width: 450px;
        height: 300px;
      }

      #content img {
        pointer-events: none;
        position: absolute;
        width: 450px;
        height: 300px;
        z-index: -1;
      }
`

This works fine, gives the desired result.

**Comment by tzstone** [comment]
```html
<div id='wrapper'>
   <div class='content'>
     ....
   </div>
</div>

<style>
#wrapper {
transform: scale(0.5)
}
</style>

<script>
const el = document.querySelector('.content') // or query '#wrapper'
html2canvas(el, {
  onclone: (clonedDocument) => {
    // reset the element which has set css 'transform: scale(xxx)'
    clonedDocument.querySelector('#wrapper').style.transform = 'scale(1)'
  }
})
</script>
```
